lib/interface_quda.cpp

```diff
@@ -2205,70 +2205,123 @@ void cloverQuda(void *h_out, void *h_in, QudaInvertParam *inv_param, QudaParity
 
 void eigensolveQuda(void **host_evecs, double _Complex *host_evals, QudaEigParam *eig_param)
 {
+  if (!initialized) errorQuda("QUDA not initialized");
+
   profileEigensolve.TPSTART(QUDA_PROFILE_TOTAL);
   profileEigensolve.TPSTART(QUDA_PROFILE_INIT);
 
-  // Transfer the inv param structure contained in eig_param
+  // Transfer the inv param structure contained in eig_param.
+  // This will define the operator to be eigensolved.
   QudaInvertParam *inv_param = eig_param->invert_param;
 
-  if (!initialized) errorQuda("QUDA not initialized");
-
-  pushVerbosity(inv_param->verbosity);
-  if (getVerbosity() >= QUDA_DEBUG_VERBOSE) {
-    printQudaInvertParam(inv_param);
-    printQudaEigParam(eig_param);
+  // Specific changes to the default invert param for the eigensolver.
+  //------------------------------------------------------------------
+  // QUDA's device routines require UKQCD gamma basis. QUDA will
+  // automatically rotate from the Degrand-Rossi basis on the host, to UKQCD
+  // on the device, and back to this basis upon completion.
+  inv_param->gamma_basis = QUDA_DEGRAND_ROSSI_GAMMA_BASIS;
```

Comment:
If one is interfacing with an application that requires UKQCD ordering or another ordering completely, hardcoding this to DeGrand-Rossi doesn't seem like the right approach.  Shouldn't it just use what value of `gamma_basis` the user has passed in `QudaEigParam::QudaInvertParam`?

Reply:
All works well without this, so removed. 7e005b04f

Reply:
7e005b04f

Reply:
All works well without this, so removed. 7e005b04f